### PR TITLE
Additional psalm param typehint for orderBy argument in findBy method

### PR DESCRIPTION
**Pull request by KartaviK**
Because of `findBy([], ['somField' => 'DESC']...);`

### CODE REVIEW

File: lib/Doctrine/ORM/EntityRepository.php [hunk 1]
```diff
@@ -185,7 +185,7 @@ public function findAll()
      * @param int|null $offset
      *
      * @psalm-param array<string, mixed> $criteria
-     * @psalm-param list<string>|null $orderBy
+     * @psalm-param list<string>|array<string, string>|null $orderBy
```
Comment (greg0ire) [inline lib/Doctrine/ORM/EntityRepository.php]: Good catch, my bad! Should `list<string>` be removed? Also, `EntityPersister::loadAll()` seems to have the same issue, and `EntityPersister::getSelectSQL()` could be improved.

Comment (KartaviK) [reply lib/Doctrine/ORM/EntityRepository.php]: If `list<string>` can not be, so it can be removed

Comment (KartaviK) [inline lib/Doctrine/ORM/EntityRepository.php]:
```suggestion
     * @psalm-param array<string, string>|null $orderBy
```

**Comment by KartaviK** [comment]
@greg0ire Updated!

File: lib/Doctrine/ORM/EntityRepository.php (hunk 1 above)
Comment (KartaviK) [reply lib/Doctrine/ORM/EntityRepository.php]: Should i update `EntityPersister::loadAll()`?

Comment (greg0ire) [reply lib/Doctrine/ORM/EntityRepository.php]: Yes please, and `EntityPersister::getSelectSQL()` too please :pray:

**Review by greg0ire — APPROVED** [review]

**Comment by greg0ire** [comment]
@orklah @vladyslavstartsev @simonberger please review, and please also tell me if you would rather I don't ping you for such reviews.

**Comment by orklah** [comment]
I don't mind being pinged :)

Ideally, this [line](https://github.com/doctrine/persistence/blob/2.1.x/lib/Doctrine/Persistence/ObjectRepository.php#L42) in doctrine/persistence should be updated to match `array<string, string>` accordingly as well.

It could be the occasion to update some other case of $orderBy as well like in AbstractEntityPersister::getHash

**Comment by greg0ire** [comment]
Maybe we could go even further, see https://github.com/doctrine/persistence/pull/165 ?
I looked for parameters called `$orderBy` in the whole project, here is the list of signatures that should be updated:
- `EntityRepository::findBy` (wrong)
- `EntityRepository::findOneBy` (imprecise)
- `AbstractEntityPersister::getHash` (imprecise)
- `EntityPersister::getSelectSQL` (imprecise)
- `EntityPersister::load` (imprecise)
- `EntityPersister::loadAll` (wrong)

Maybe this PR should only fix what's wrong so `EntityRepository::findBy` and `EntityPersister::loadAll`, and the rest should be improved in a PR targeting 2.9.x

**Review by greg0ire — DISMISSED** [review]

**Comment by greg0ire** [comment]
I updated the PR according to that plan.

**Review by greg0ire — APPROVED** [review]

**Comment by greg0ire** [comment]
Thanks @KartaviK !

**Pull request merged**